AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -327,23 +335,38 @@ - (void)beginUpdates
 
 - (void)endUpdates
 {
-  [self endUpdatesWithCompletion:NULL];
+  [self endUpdatesAnimated:YES completion:nil];
 }
 
-- (void)endUpdatesWithCompletion:(void (^)(BOOL))completion
+- (void)endUpdatesAnimated:(BOOL)animated completion:(void (^)(BOOL))completion
 {
   _batchUpdateCounter--;
 
   if (_batchUpdateCounter == 0) {
-    [_delegate dataControllerBeginUpdates:self];
+    LOG(@"endUpdatesWithCompletion - beginning");
+
+    [_editingTransactionQueue addOperationWithBlock:^{
```

Comment:
In this patch or a followup, it would be worth adding short comments describing here and in the endUpdates delegate call case why it is important to queue them.

Here, the idea is that a runloop turn needs to take place (is this for UIKit or some other reason?)  For the end call, it's necessary to allow the running of main thread delegate call blocks that were scheduled by calling the _pendingEditCommandBlocks.  Since the current method is operating on main, it would otherwise always be able to call the delegate method before the queued ones completed — even if execution waited on each block.